pkg/node/kubelet.go

```diff
@@ -84,3 +87,74 @@ func StartKubeProxy(cfg *config.MicroshiftConfig) error {
 
 	return nil
 }
+
+const (
+	// proxy component name
+	component = "kubeproxy"
+)
+
+type ProxyOptions struct {
+	options *kubeproxy.Options
+}
+
+func NewKubeProxyServer(cfg *config.MicroshiftConfig) *ProxyOptions {
+	s := &ProxyOptions{}
+	s.configure(cfg)
+	return s
+}
+
+func (s *ProxyOptions) Name() string           { return component }
+func (s *ProxyOptions) Dependencies() []string { return []string{"kube-apiserver"} }
+
+func (s *ProxyOptions) configure(cfg *config.MicroshiftConfig) error {
+	if err := config.KubeProxyConfig(cfg); err != nil {
+		logrus.Infof("Failed to create a new kube-proxy configuration: %v", err)
+		return err
+	}
+	args := []string{
+		"--logtostderr=" + strconv.FormatBool(cfg.LogDir == "" || cfg.LogAlsotostderr),
+		"--alsologtostderr=" + strconv.FormatBool(cfg.LogAlsotostderr),
+		"--v=" + strconv.Itoa(cfg.LogVLevel),
+		"--vmodule=" + cfg.LogVModule,
+	}
+	cmd := &cobra.Command{
+		Use:          component,
+		Long:         component,
+		SilenceUsage: true,
+		RunE:         func(cmd *cobra.Command, args []string) error { return nil },
+	}
+	if cfg.LogDir != "" {
+		args = append(args, "--log-file="+filepath.Join(cfg.LogDir, "kube-proxy.log"))
+	}
+
+	opts := kubeproxy.NewOptions()
+	opts.ConfigFile = cfg.DataDir + "/resources/kube-proxy/config/config.yaml"
+	opts.Complete()
+	s.options = opts
+	cmd.SetArgs(args)
+
+	if err := cmd.ParseFlags(args); err != nil {
+		logrus.Fatalf("failed to parse flags:%v", err)
+	}
+	logrus.Infof("starting kube-proxy, args: %v", args)
+	return nil
+}
+
+func (s *ProxyOptions) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+
+	defer close(stopped)
+	// run readiness check
+	// go func() {
+	// 	healthcheckStatus := util.RetryInsecureHttpsGet("https://127.0.0.1:10250/healthz")
```

Comment:
The default port is 10256.

Reply:
Yeah I tested with 10256 too but the server is not up

Reply:
it should work
```console
# curl http:netstat -anp |grep 10256
tcp6       0      0 :::10256                :::*                    LISTEN      3743888/_output/bin
# curl http://127.0.0.1:10256/healthz
{"lastUpdated": "2021-09-07 14:47:41.024941778 -0400 EDT m=+85.884851029","currentTime": "2021-09-07 14:47:41.024941778 -0400 EDT m=+85.884851029"}
```

Reply:
Using `http` works